dashboard/app/controllers/reports_controller.rb

```diff
@@ -254,9 +255,33 @@ def monthly_metrics
 
   def pd_progress
     authorize! :read, :reports
-    @script = Script.find_by(name: params[:script] || 'K5PD')
+    script = Script.find_by(name: params[:script] || 'K5PD')
     # Get all users with any activity in the script
-    @users = Activity.where(level_id: @script.levels.map(&:id)).map(&:user_id).uniq.map{|id|User.find(id)}
+    users = Activity.where(level_id: script.levels.map(&:id)).map(&:user_id).uniq.map{|id|User.find(id)}
+    headers = nil
+    data = users.map do |user|
+      row = {}
+      row.merge!({
+                     :'ID' => user.id,
+                     :'Ops First Name' => user.ops_first_name,
+                     :'Ops Last name' => user.ops_last_name,
+                     :'Email' => user.email,
+                     :'District' => user.district || 'None'
+                 })
+      user_progress = summarize_user_progress(script, user)
+      percent = percent_complete(script, user)
+      script.stages.each do |stage|
+        levels = Hash[stage.script_levels.map(&:level).map do |level|
+          ["<a href='#{level_path(level.id)}'>#{level.id}</a>", (progress = user_progress[:levels][level.id]) && progress[:status] == 'perfect' ? '1' : '0']
+                      end]
```

Comment:
Lint/BlockAlignment: `end` at 276, 22 is not aligned with `levels = Hash[stage.script_levels.map(&:level).map do |level|` at 274, 8